src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,36 @@
-import React from 'react';
+import React, { ChangeEvent } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  function handleChangeQuery(newQuery: string) {
+    const normalizedQuery = newQuery.toLowerCase();
+
+    if (normalizedQuery !== query) {
+      dispatch(filterSlice.actions.changeQuery(normalizedQuery));
+    }
+  }
+
+  function handleChangeStatus(event: ChangeEvent<HTMLSelectElement>) {
+    if (status !== event.currentTarget.value) {
+      dispatch(
+        filterSlice.actions.changeStatus(event.currentTarget.value as Status),
+      );
+    }
+  }
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select data-cy="statusSelect" onChange={handleChangeStatus}>
```

Comment:
The `select` element should have a `value` attribute to make it a controlled component. This ensures that the UI reflects the current state accurately. Add `value={status}` to the `select` element to bind it to the current status from the state.